platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -125,20 +259,52 @@ public synchronized boolean putIfEqual(final long index, final long oldValue, fi
     @Override
     public void writeToFile(final Path newFile) throws IOException {
         // finish writing to current file
-        fileChannel.force(true);
+        currentFileChannel.force(true);
         // if new file is provided then copy to it
-        if (!file.equals(newFile)) {
-            try (final FileChannel fc =
-                    FileChannel.open(newFile, StandardOpenOption.CREATE, StandardOpenOption.WRITE)) {
-                fileChannel.position(0);
-                MerkleDbFileUtils.completelyTransferFrom(fc, fileChannel, 0, fileChannel.size());
-            }
+        try (final FileChannel fc = FileChannel.open(newFile, StandardOpenOption.CREATE, StandardOpenOption.WRITE)) {
+            // write header
+            writeHeader(fc);
+            // write data
+            writeLongsData(fc);
+            fc.force(true);
         }
     }
 
-    /** No-op as we override writeToFile directly */
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    protected void writeLongsData(final FileChannel fc) {}
+    protected void writeLongsData(final FileChannel fc) throws IOException {
+        final int totalNumOfChunks = calculateNumberOfChunks(size());
+        final long currentMinValidIndex = minValidIndex.get();
+        final int firstChunkWithDataIndex = (int) currentMinValidIndex / numLongsPerChunk;
+        final ByteBuffer transferBuffer = transferBufferThreadLocal.get();
+        // write data
```

Comment:
Usually, a comment like this indicates the block wants to be a method with a descriptive name.  Not a necessity, just something to think about.